test/parallel_api/algorithm/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -17,107 +17,128 @@
 #include <oneapi/dpl/algorithm>
 #include <oneapi/dpl/iterator>
 
-#include <iostream>
-
 #include "support/test_config.h"
 #include "support/utils.h"
+#include "../binary_search_utils.h"
 
 #if TEST_DPCPP_BACKEND_PRESENT
+#    include <CL/sycl.hpp>
 
-#include <CL/sycl.hpp>
+using namespace oneapi::dpl::execution;
+#endif
+using namespace TestUtils;
 
-void test_on_device()
+struct test_binary_search
 {
-    bool correctness_flag =true;
-
-    //Test case 1
-    cl::sycl::buffer<uint64_t, 1> _key_buf{ cl::sycl::range<1>(10) };
-    cl::sycl::buffer<uint64_t, 1> _val_buf{ cl::sycl::range<1>(5) };
-    cl::sycl::buffer<uint64_t, 1> _res_buf{ cl::sycl::range<1>(5) };
-    { 
-        auto key_buf = _key_buf.template get_access<cl::sycl::access::mode::read_write>();
-	auto val_buf = _val_buf.template get_access<cl::sycl::access::mode::read_write>();
-	auto res_buf = _res_buf.template get_access<cl::sycl::access::mode::read_write>();
-     
-	// Initialize data
-	key_buf[0] = 0; key_buf[1] = 2; key_buf[2] = 2; key_buf[3] = 2; key_buf[4] = 3;
-	key_buf[5] = 3; key_buf[6] = 3; key_buf[7] = 3; key_buf[8] = 6; key_buf[9] = 6;
-	
-	val_buf[0] = 0; val_buf[1] = 2; val_buf[2] = 4; val_buf[3] = 7; val_buf[4] = 6;
+    template <typename Accessor1, typename Size>
+    void
+    check_values(Accessor1 result, Size n)
+    {
+        int num_values = n * .01 > 1 ? n * .01 : 1; // # search values expected to be << n
+        for (int i = 0; i != num_values; ++i)
+        {
+            if (i == 0)
+            {
+                EXPECT_TRUE(result[i] == true, "wrong effect from binary_search");
+            }
+            else
+            {
+                EXPECT_TRUE(result[i] == i % 2, "wrong effect from binary_search");
+            }
+            // clean result for next test case
+            result[i] = 0;
```

Comment:
I would not expect that cleaning of the results happens when we check them. Let's use fill algorithm after each call to make the test more straightforward. Another option is to rename the function, e.g. to check_and_clean.